src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -169,6 +170,59 @@ public void InitializeService(ServiceHost serviceHost)
         }
 
         /// <summary>
+        /// Handles a request to execute a string and return the result
+        /// </summary>
+        internal Task HandleExecuteAndReturnResultRequest(ExecuteAndReturnResultParams executeParams,
+            RequestContext<ExecuteAndReturnResultResult> requestContext)
+        {
+            Func<string, Task> queryCreateFailureAction = message => requestContext.SendError(message);
+            ExecuteStringParams newParams = new ExecuteStringParams
+            {
+                Query = executeParams.QueryString,
+                OwnerUri = executeParams.OwnerUri
+            };
+
+            ResultOnlyContext<ExecuteAndReturnResultResult> newContext = new ResultOnlyContext<ExecuteAndReturnResultResult>(requestContext);
+
+            // handle sending event back when the query completes
+            Query.QueryAsyncEventHandler queryComplete = async q =>
+            {
+                int rowCount = 0;
+                try
+                {
+                    rowCount = Convert.ToInt32(q.Batches[0].ResultSets[0].RowCount);
+                }
+                catch(OverflowException e)
+                {
+                    await requestContext.SendError(e);
```

Comment:
I wanted this to be the simplest query execution possible so i wanted it to be limited to 1 batch 1 resultset, having multiple batches and results world complicate the response more than what I was looking for. But if we think this is something that should be done I can implement something like that.

On the topic of this try catch, @benrr101 can explain a bit more, but basically there is a worry that if the client requests a rowcount that is too large it could take up all the memory, so limiting this value to a int makes sense. I meant this try catch to be the point that it checks if there are too many rows to safely load into memory. Although I agree the error sent back should be better. 

The other way I guess this could be done to avoid try catch is to explicit check if the long is greater than 2^31 - 1 and send back an error if it is.